compiler/noirc_evaluator/src/ssa/ssa_gen/context.rs

```diff
@@ -370,26 +336,9 @@ impl<'a> FunctionContext<'a> {
         match result_type {
             NumericType::Signed { bit_size } => {
                 match operator {
-                    BinaryOpKind::Add | BinaryOpKind::Subtract => {
-                        // Result is computed modulo the bit size
-                        let result = self.builder.insert_truncate(result, bit_size, bit_size + 1);
-                        let result = self.insert_safe_cast(
-                            result,
-                            NumericType::unsigned(bit_size),
-                            location,
-                        );
-
-                        self.check_signed_overflow(result, lhs, rhs, operator, bit_size, location);
```

Comment:
The docstring of this method, `check_overflow`, talks extensively about delegating to `check_signed_overflow`, which is now removed below.